server/src/main/java/org/opensearch/action/support/clustermanager/MasterThrottlingRetryListener.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.support.clustermanager;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.ActionResponse;
+import org.opensearch.action.bulk.BackoffPolicy;
+import org.opensearch.cluster.metadata.ProcessClusterEventTimeoutException;
+import org.opensearch.cluster.service.MasterTaskThrottlingException;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.threadpool.Scheduler;
+import org.opensearch.transport.TransportException;
+
+import java.util.Iterator;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * ActionListener for retrying the Throttled master tasks.
+ * It schedules the retry on the Throttling Exception from master node and
+ * delegates the response if it receive response from master.
+ *
+ * It uses ExponentialEqualJitterBackoff policy for determining delay between retries.
+ */
+public class MasterThrottlingRetryListener<Request extends ClusterManagerNodeRequest<Request>, Response extends ActionResponse>
+    implements
+        ActionListener<Response> {
+
+    private static final Logger logger = LogManager.getLogger(MasterThrottlingRetryListener.class);
+
+    /**
+     * Base delay in millis.
+     */
+    private final int BASE_DELAY_MILLIS = 10;
+
+    /**
+     * Maximum delay in millis.
+     */
+    private final int MAX_DELAY_MILLIS = 5000;
+
+    private long totalDelay;
+    private final Iterator<TimeValue> backoffDelay;
+    private final ActionListener<Response> listener;
+    private final Request request;
+    private final Runnable runnable;
+    private final String actionName;
+    private final boolean localNodeRequest;
+
+    private static ScheduledThreadPoolExecutor scheduler = Scheduler.initScheduler(Settings.EMPTY);
```

Comment:
retries will be scheduled on a different threadpool than the original thread pool?

Reply:
This class has been removed and reused the RetryableAction for retries.